test/models/product_test.rb

```diff
@@ -1,7 +1,36 @@
 require 'test_helper'
 
 class ProductTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test 'Test valid product' do
+    product = create(:product)
+    assert product.valid?
+  end
+
+  test 'Invalid without name set' do
+    product = create(:product)
+    product.name = nil
+    refute product.valid?, 'product valid without name'
+    assert_not_nil product.errors[:name], 'no validation error for name'
+  end
+
+  test 'Invalid without available set' do
```

Comment:
available is a boolean so it should have a database default and not be validated